### PR TITLE
 Color mixer channels if they are made by a colored track

### LINKED ISSUE #5778: Assign track color to mixer automatically.
### Summary of improvements
Assign the track color to the mixer channel automatically.

#### Justification
Currently you can assign a track to a channel of the mixer, the channel will be created automatically, but the color of the track is not assigned to the channel and you have to manually select the color of the channel, which is a bit cumbersome and impractical.

#### Sketch

Here is an image of the problem.
Opcion para asignar color al mezclador
The track has a color assigned but the channel it was assigned to does not.

### PR DESCRIPTION
**Pull request by russiankumar**
~**Don't merge yet.** There's discussion going on in #5778 on whether this is a needed feature, I'll merge this/work a bit more on it if consensus is reached.~

Closes #5778.

This PR implements the functionality to color the new mixer tracks generated by the 'Assign to new FX channel' option.

#### Demo

capture5

### CODE REVIEW

**Comment by Gabrielxd195** [comment]
I just tested the upgrade you made and I see that everything works fine. now the color is assigned automatically. Thank you

**Comment by russiankumar** [comment]
Great, thanks for testing!

Could anybody glance at the code for this? It's 13 lines, should (hopefully) be quick.

Note about the style: As discussed in #5765, I have used the new code style here. Please refer to the conversation over there for why it was done.

File: src/tracks/SampleTrack.cpp [hunk 1]
```diff
@@ -1169,8 +1169,10 @@ void SampleTrackWindow::modelChanged()
 void SampleTrackView::createFxLine()
 {
 	int channelIndex = gui->fxMixerView()->addNewChannel();
+	auto channel = Engine::fxMixer()->effectChannel(channelIndex);
 
-	Engine::fxMixer()->effectChannel(channelIndex)->m_name = getTrack()->name();
+	channel->m_name = getTrack()->name();
+	if (getTrack()->useColor()) channel->setColor (getTrack()->color());
```
Comment (IanCaio) [inline src/tracks/SampleTrack.cpp:1175]: Same as above

Comment (IanCaio) [reply src/tracks/SampleTrack.cpp:1175]: Something weird happened, my two other comments didn't get submitted with the review

File: include/FxMixer.h [hunk 2]
```diff
@@ -73,7 +73,13 @@ class FxChannel : public ThreadableJob
 		void unmuteForSolo();
 
 
-		// TODO C++17 and above: use std::optional insteads
+		void setColor (QColor newColor)
```
Comment (IanCaio) [inline include/FxMixer.h:76]: Now that you have `setColor` for the `FxMixer`, you could use that on the `FxLine` code (`changeColor` and `randomColor`) for consistency.

File: src/tracks/InstrumentTrack.cpp [hunk 3]
```diff
@@ -1098,8 +1098,10 @@ InstrumentTrackWindow * InstrumentTrackView::topLevelInstrumentTrackWindow()
 void InstrumentTrackView::createFxLine()
 {
 	int channelIndex = gui->fxMixerView()->addNewChannel();
+	auto channel = Engine::fxMixer()->effectChannel(channelIndex);
 
-	Engine::fxMixer()->effectChannel( channelIndex )->m_name = getTrack()->name();
+	channel->m_name = getTrack()->name();
+	if (getTrack()->useColor()) channel->setColor (getTrack()->color());
```
Comment (IanCaio) [inline src/tracks/InstrumentTrack.cpp:1104]: Minor code convention issue (even one liners require explicit blocks):
```suggestion
	if (getTrack()->useColor()) { channel->setColor (getTrack()->color()); }
```

That's what was referred earlier as "Same as above", for some reason only the last comment was submitted.

File: include/FxMixer.h (hunk 2 above)
Comment (russiankumar) [reply include/FxMixer.h:76]: That's a little bit beyond this PR... but since I'm the author of those, I'll do that as well.

**Comment by superpaik** [comment]
Tested ok.

File: src/tracks/InstrumentTrack.cpp (hunk 3 above)
Comment (russiankumar) [reply src/tracks/InstrumentTrack.cpp]: Didn't know this, thanks for pointing it out.

File: src/tracks/SampleTrack.cpp (hunk 1 above)
Comment (russiankumar) [reply src/tracks/SampleTrack.cpp]: Strange... Anyways, made the changes.

**Comment by russiankumar** [comment]
Thanks for the testing, @Gabrielxd195 and @superpaik! If @IanCaio approves, I think this will be good to go.

**Review by DomClark — APPROVED** [review]

**Review by IanCaio — APPROVED** [review]

**Comment by russiankumar** [comment]
It appears that there is quite some discussion over at #5780. I’ll hold this one out until there’s a consensus, and if everyone agrees to have this merge, I’ll merge it.

**Review by Spekular — APPROVED** [review]
LGTM and not opposed to this implementation (see https://github.com/LMMS/lmms/issues/5778#issuecomment-725961472)

**Comment by russiankumar** [comment]
Alright, it seems consensus in favour of this has been reached. Merging this.

**Pull request merged**